ironfish-rust/src/keys/mod.rs

```diff
@@ -132,17 +132,13 @@ impl SaplingKey {
 
     /// Load a key from a string of hexadecimal digits
     pub fn from_hex(value: &str) -> Result<Self, IronfishError> {
-        match hex_to_bytes(value) {
-            Err(_) => Err(IronfishError::InvalidPaymentAddress),
-            Ok(bytes) => {
-                if bytes.len() != 32 {
-                    Err(IronfishError::InvalidPaymentAddress)
-                } else {
-                    let mut byte_arr = [0; 32];
-                    byte_arr.clone_from_slice(&bytes[0..32]);
-                    Self::new(byte_arr)
-                }
-            }
```

Comment:
This was swallowing errors from `hex_to_bytes`, I am ok with reverting this change, but I think the new version is more explicit